rtbkit/core/router/filters/static_filters.cc

```diff
@@ -82,6 +82,17 @@ applyExchangeFilter(FilterState& state, const ConfigSet& result) const
 
 void
 SegmentsFilter::
+fillFilterReasons(FilterState& state, ConfigSet& beforeFilt,
+                  ConfigSet& afterFilt, const std::string & segment) const {
+
+    // Some Magic to get all the filtered out configs by this segment.
+    FilterState::FilterReasons& reasons = state.getFilterReasons();
+    reasons[segment] = beforeFilt ^ (beforeFilt & afterFilt);
```

Comment:
So I was thinking it over during the week-end and I think the original version of the code is correct and we should keep it. The reason is that the sub-filter is not required to return a strict subset of the current filter because the current filter set is not fed to it.

Sorry about the flip-flop :)